src/icp_v2.cc

```diff
@@ -136,19 +136,16 @@ icp_common_t::getOpCode() const
 
 /* ICPState */
 
-ICPState::ICPState(icp_common_t &aHeader, HttpRequest *aRequest):
+ICPState::ICPState(icp_common_t &aHeader, const HttpRequestPointer &aRequest):
```

Comment:
If you recall from previous discussions, I think we should use Foo::Pointer[^1] whenever that name has to be visible (for other valid reasons).

[^1]: Instead of a FooPointer _forward_ declaration.

This suggestion illustrates the change but this change request applies to most other (if not all other) PR changes as well.

```suggestion
ICPState::ICPState(icp_common_t &aHeader, const HttpRequest::Pointer &aRequest):
```

I do not insist on these changes.